ddsrecorder_participants/src/cpp/recorder/mcap/McapHandler.cpp

```diff
@@ -853,7 +861,12 @@ mcap::ChannelId McapHandler::create_channel_id_nts_(
     // Create new channel
     mcap::KeyValueMap metadata = {};
     metadata[QOS_SERIALIZATION_QOS] = serialize_qos_(topic.topic_qos);
-    mcap::Channel new_channel(topic.m_topic_name, "cdr", schema_id, metadata);
+    metadata[ROS2_TYPES] =
+            configuration_.ros2_types ? topic.m_topic_name.rfind("rt/",
+                    0) == 0 ? "true" : "false" : "false";
+    std::string topic_name =
+            configuration_.ros2_types ? utils::demangle_if_ros_topic(topic.m_topic_name) : topic.m_topic_name;
```

Comment:
Isn't the same condition being checked twice? If so, compare the string returned by `utils::demangle_if_ros_topic` to know if demangling was performed.

Reply:
I think the condition you added should be the other way around (if both are equal, ros2 types should be false right?).

Reply:
True